subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperLoggingIntegrationTest.groovy

```diff
@@ -31,4 +34,39 @@ task emptyTask
         then:
         result.output.empty
     }
+
+    def "wrapper logs when there is a problem setting permissions"() {
+        given: "malformed distribution"
+        // Repackage distribution with bin/gradle removed so permissions cannot be set
+        def tempUnzipDir = temporaryFolder.createDir("temp-unzip")
+        distribution.binDistribution.unzipTo(tempUnzipDir)
+        distribution.binDistribution.delete()
+        assert tempUnzipDir.file("gradle-${distribution.version.version}", "bin", "gradle").delete()
+        tempUnzipDir.zipTo(distribution.binDistribution)
+        prepareWrapper(distribution.binDistribution.toURI())
+
+        when:
+        def failure = wrapperExecuter
+            .withTasks("emptyTask")
+            .withStackTraceChecksDisabled()
+            .run()
+
+        then:
+        failure.output.contains("Could not set executable permissions")
+    }
+
+    def "wrapper logs when unzipping distribution fails"() {
+        given: "Empty zip file"
+        prepareWrapper(distribution.binDistribution.toURI())
+        distribution.binDistribution.text = ""
+
+        when:
+        ExecutionFailure failure = wrapperExecuter
+            .withTasks("emptyTask")
+            .withStackTraceChecksDisabled()
+            .runWithFailure()
+
+        then:
+        failure.output.contains("Could not unzip")
```

Comment:
:o: We should probably test that "setting the permission" error message doesn't show up. Having two different error messages would be confusing to users.

Reply:
Done